data/src/tensor.rs

```diff
@@ -830,17 +838,75 @@ impl Tensor {
                     };
                 };
             }
-            n!(u8);
-            n!(u16);
-            n!(u32);
-            n!(u64);
-            n!(i8);
-            n!(i16);
-            n!(i32);
-            n!(i64);
-            n!(f16);
-            n!(f32);
-            n!(f64);
+
+            if !dt.is_quantized() && !self.datum_type().is_quantized() {
+                n!(u8);
+                n!(u16);
+                n!(u32);
+                n!(u64);
+                n!(i8);
+                n!(i16);
+                n!(i32);
+                n!(i64);
+                n!(f16);
+                n!(f32);
+                n!(f64);
+            }
+            //FIXME : round properly (and maybe try to reduce code size / improve the macro ?)
+            use num_traits::AsPrimitive;
+            let float_convert = ((self.datum_type().is_float() || dt.is_float())
+                && (self.datum_type().is_quantized() || dt.is_quantized()))
+                || (self.datum_type().is_quantized() && dt.is_quantized());
+
+            let (s_zp, s_scale) = self.datum_type().zp_scale();
+            let (d_zp, d_scale) = dt.zp_scale();
+            macro_rules! q_n {
```

Comment:
Is it better to do one big macro that you call once, or a simpler macro you call multiple times ?